### ISSUE TITLE
Support for regex or globs in interface name (Network module)

### ISSUE THREAD

**Issue by mil-ad** [polybar/polybar#2551]
It'd be nice if we could use globs for interface names in the network module. My network link has different names on different machines but ideally I'd like to use the same config file. It'd be nice if I could use something like `interface = en*`

**Comment by patrick96** [comment]
There is already a new `interface-type` setting coming in the next release, where you can select either `wired` or `wireless`. Would that address your use-case?

**Comment by mil-ad** [comment]
so does that pick up the _first_ wired interface? What happens if there multiple links? I normally just have one wired interface but its name is different based on computer or the ethernet dongle I use.

**Comment by patrick96** [comment]
It will choose the first one it finds. If you have multiple wired interfaces, your glob `en*` will probably match all of them as well.

**Comment by mil-ad** [comment]
True. As long as this new `interface-type` does not require explicit interface name I think I'll be fine but globs are probably more flexible? They also match how systemd-network deals with the same issue. An example of `/etc/systemd/network/20-wired.network`:

```
[Match]
Name=en*

[Network]
DHCP=ipv4
```

**Comment by patrick96** [comment]
Yeah, globs or regexes are definitely more flexible. If anyone wants to implement this, PRs are welcome. Make sure to also comment here, so that we know you are working on it.

Either globs or regex are fine. Using `glob` (`man 3 glob`) is more user-friendly but is tied to the filesystem (cannot write tests for it) and I'm not sure how safe it is to pass user-supplied data to it (can it crash the process?). Regex is more flexible and is directly supplied in the standard library.

**Comment by mil-ad** [comment]
Great, I'll have a go at this weekend

**Comment by patrick96** [comment]
Perfect :+1:

**Comment by mil-ad** [comment]
Sorry I completely forgot about this!

**Comment by madhavpcm** [comment]
Is this being implemented ? Im willing to work on this.

**Comment by patrick96** [comment]
There hasn't been any PRs for this feature that I'm aware. @bazuin-32, @madhavpcm You can decide among yourselves who works on this.

**Comment by rossbrierley** [comment]
Hi, I noticed this issue about a week ago, it seems like this issue is still unsolved. I would like to try and fix it!
